### ISSUE TITLE
Aligning Word output styles to pre-existing Word template?

### ISSUE THREAD

**Issue by mhGLEIF** [jelovirt/com.elovirta.ooxml#26]
Thanks for the great work so far! :1st_place_medal: 

I need to do one last tweak to the plugin before I can put in into production:

I have a Word template with highly customized styles.

These aren't always the styles used by the DITA-Word converter.

Which angle is best/simplest to start from in order to align them?

Edit the Word template to match whatever heading/numberedlist/bulletlist/emphasis etc. are already mapped in the plugin? Is there a full listing of these I can see?

Or edit some config file in the plugin (if so, which one please?) to match my existing Word setup (it is not too complex, it has numbered headings 1-9, bullet list, normal, I use the default Word HTML Code a lot... and that's almost everything)?

**Comment by jelovirt** [comment]
See https://github.com/jelovirt/com.elovirta.ooxml/wiki/Templates and https://github.com/jelovirt/com.elovirta.ooxml/wiki/Extending

There is no list of element to style mapping. We could add a mapping to the wiki documentation, but the problem is that the listing will quickly become out of date.

**Comment by mhGLEIF** [comment]
Shouldn't we aim to have a fixed, 1:1 mapping of DITA element --> Word style?

**Comment by jelovirt** [comment]
No, because the mapping is always specific to Word template. We can create mapping from e.g. `ul/li` -> `Bulleted list 1`, but then you will have templates that use `Company unordered list 1` style for that and you will need to override the default mapping.

**Comment by mhGLEIF** [comment]
This is a very fair point. I would still welcome a default mapping with a caveat that it will need updating for every user's template - it would give a useful starting point especially for anyone implementing this as a solo project :-)

**Comment by jelovirt** [comment]
A default mapping already exists, but it's just not documented in the wiki, it's only in the stylesheets. Them mapping basically works so that for all structures that have a mapping, that's the one that's used; all other elements will fall back to using the default style.

**Comment by mhGLEIF** [comment]
Thanks, that's clear; so some DITA elements may have no specific Word style target?

I would propose mapping as much as possible in basic DITA (ideally everything) to one of the default Word styles (no matter how bad ;-) ) just to make it easier for implementers to find something and change it in the publishing pipeline.

**Comment by jelovirt** [comment]
Yes, *all* elements will map to *some* style. In most cases it's `Body Text` for block-level element and and no style for inline-level elements.

**Comment by mhGLEIF** [comment]
That's the problem :-) I'd like to find a table somewhere with a "comprehensive" (as practical) set of DITA styles that I can just change to fit my local template, rather than have to add new rows.

I'm happy to propose something if it helps...

**Comment by jelovirt** [comment]
We can document the current mapping in the wiki, but like I said, it will be out of date quickly. And, because *all* styles are read from the Word template, the mapping is always specific to the template. Thus, if you don't use the default template that comes with the plugin, you always have to map DITA structures to styles. So there is no way you can skip that step.

**Comment by mhGLEIF** [comment]
I think we agree - I don't want to _skip_ the step of mapping, it's just about _how_ to do the mapping.

If there's already a fairly comprehensive mapping (even though it will always be wrong!) I can just edit the existing links instead of creating them from zero.

Maybe I didn't understand correctly?

**Comment by mhGLEIF** [comment]
As an implementer it's much less overhead for me to generate a couple of examples, compare the output with the current configs, and modify < - - > test a few times to get the correct setup.

**Comment by jelovirt** [comment]
Yes, I though you were talking about documentation, when you wanted to have a file with the mapping to edit.

You have to create a stylesheet to change the mapping. An example of such a customization should be in the wiki.

**Comment by mhGLEIF** [comment]
If most items on either end of the pipeline default to the same thing it's more work to track down what changes to make - but with a default mapping, even if the result is _wrong_ at least it's _different_...

**Comment by mhGLEIF** [comment]
OK, I'lll take a look at the wiki and try to figure it out :-)

**Comment by mhGLEIF** [comment]
I will be honest - I am not understanding how to do this mapping. I can't understand how to read "this DITA element becomes that Word style" in the xsl (sorry!)...

Specifically, I can't find `block-style` and `inline-style` in https://github.com/jelovirt/com.elovirta.ooxml/blob/master/docx/word/document.xsl

**Comment by jelovirt** [comment]
That file doesn't contain those mappings. See e.g. https://github.com/jelovirt/com.elovirta.ooxml/blob/master/docx/word/document.topic.xsl#L223 or https://github.com/jelovirt/com.elovirta.ooxml/blob/master/docx/word/document.pr-d.xsl#L21

**Comment by rahulraykor** [comment]
Can't we do this like:
In topics/sections/paragraph there will be a element which will contain the style id.
If any custom template is used for conversion we can assign that style (specified in dita xml) to the element and if default template is used then we will use the style id's in default template.

Please note the person who is converting the document using template, it should be his responsiblity to make sure that all the styles are present in template he is using.

what is your thinking @jelovirt @mhGLEIF ?
